### ISSUE TITLE
Magento 2.3.0 contains references in frontend html to tinymce 4 wysiwyg editor

### ISSUE THREAD

**Issue by hostep** [magento/magento2#21766]
### Preconditions (*)
1. PHP 7.2.15
2. Magento 2.3.0

### Steps to reproduce (*)
1. Install a vanilla Magento 2.3.0
2. Open the homepage on the frontend
3. Look at the generated html

### Expected result (*)
1. Not seeing a reference to tinymce editor

### Actual result (*)
1. Seeing the following in the html:
```html
<script>
    require.config({
        map: {
            '*': {
                wysiwygAdapter: 'mage/adminhtml/wysiwyg/tiny_mce/tinymce4Adapter'
            }
        }
    });
</script>
```

### Further discussion
I believe this was incorrectly introduced in https://github.com/magento/magento2/commit/9c2ce257093d9d60742db0947c5f7907190e4721#diff-ea739bfd03b11e588d182cc408c57f5a (MAGETWO-72114)

The file `app/code/Magento/Theme/view/base/layout/default.xml` should probably be renamed to `app/code/Magento/Theme/view/adminhtml/layout/default.xml` to fix it.

**Comment by magento-engcom-team** [comment]
Hi @hostep. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@hostep do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98706` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by mrtuvn** [comment]
yes it's should be change area to adminhtml instead base. But about issue compatible backward of tinymce. I believe a lot of extensions before 2.3.0 will still implements the old way tinymce3 and will get big problem with tinymce requirejs . I don't find any links or documents about this problem compatible

**Comment by hostep** [comment]
@mrtuvn: Magento still ships with a TinyMCE v3 module, just to keep backwards compatibility with extensions: https://github.com/magento/magento2/tree/2.3.0/app/code/Magento/Tinymce3

Not quite sure what this has to do with this issue though :)

**Comment by mrtuvn** [comment]
yes i know this but when digger a bit a get confuse when try to initalize wysiwyg for tinymce
in previous version of 2.3 tinymce.init still usable but when come with 2.3.x tinymce4 come with new implement and problem will happens here. tinymce init probably not work anymore or need new way to init tinymce for both version. But this maybe wide topics and out of scope of this issue. Note here if anyone in core team can point me the link for clarify usage of tinymce

**Comment by hostep** [comment]
Back on topic for now:

Looking through some files, it seems like we have some references to the `wysiwygAdapter` in other `base` directories, so I'm wondering if we should also move those to the `adminhtml` directory as well. Or if it was the purpose of Magento to allow the TinyMCE editor to be used on the frontend?

Here are those files:
- https://github.com/magento/magento2/blob/2.3.0/app/code/Magento/Ui/view/base/web/js/form/element/wysiwyg.js
- https://github.com/magento/magento2/blob/2.3.0/app/code/Magento/Ui/view/base/requirejs-config.js
- https://github.com/magento/magento2/blob/2.3.0/app/code/Magento/Ui/view/base/templates/wysiwyg/active_editor.phtml

It looks like the most important one of those is the `Magento/Ui/view/base/web/js/form/element/wysiwyg.js` file. Looking through its history, it seems like this was introduced in the `base` directory in https://github.com/magento/magento2/commit/b025dc64afac9b4432c086c9de527a8952d0170a#diff-2732493e7f3dfb72d10e5aa2e18b9917

Let's try to include some developers who worked on this, to try to figure out if it was the purpose to expose the TinyMCE editor on the frontend as well? In that case, this issue can be closed. @AndriyNasinnyk & @cspruiell & @irenelagno

**Comment by mrtuvn** [comment]
tinymce and tinyMCE is objects exposed through browser. I think they can accessable variables. Variable tinyMCE is from tinymce4 and tinymce is from our old friend (tinymce3). From now i think we can use/require 'wysiwygAdapter' for compatible require tinymce (Work for both version 3,4). One think i don't see any example in frontend of magento that use tinymce yet. Not sure about this . Maybe in future we will use wysiwyg alias for both (tinymce or tinyMCE) instead the old way tinymce in legacy code

**Comment by Anantkprajapati** [comment]
I am working on this at #dmcdindia19

**Comment by hostep** [comment]
Let's close this, it was determined that fixing this might cause backwards incompatibilities after discussion in https://github.com/magento/magento2/pull/24093
